mender.go

```diff
@@ -15,28 +15,30 @@ package main
 
 import (
 	"bufio"
-	"encoding/json"
-	"errors"
-	"io/ioutil"
 	"os"
 	"strings"
 	"time"
 
 	"github.com/mendersoftware/log"
+	"github.com/pkg/errors"
 )
 
 type Controler interface {
-	Bootstrap() error
-	TransitionState() MenderState
+	Bootstrap() menderError
```

Comment:
Not sure if `Bootstrap()` should be a part of `Controller` interface. It feels more like client part.